core/queryalgebra/evaluation/src/main/java/org/eclipse/rdf4j/query/algebra/evaluation/ArrayBindingSet.java

```diff
@@ -57,50 +57,24 @@ public ArrayBindingSet(String... names) {
 		this.whichBindingsHaveBeenSet = new boolean[names.length];
 	}
 
-	public ArrayBindingSet(BindingSet toCopy, LinkedHashSet<String> names, String[] namesArray) {
+	public ArrayBindingSet(BindingSet toCopy, Set<String> names, String[] namesArray) {
 		assert !(toCopy instanceof ArrayBindingSet);
 
-		Set<String> toCopyBindingNames = toCopy.getBindingNames();
-
-		if (names.containsAll(toCopyBindingNames)) {
-			this.bindingNames = namesArray;
-			this.whichBindingsHaveBeenSet = new boolean[this.bindingNames.length];
-			this.values = new Value[this.bindingNames.length];
-			for (int i = 0; i < this.bindingNames.length; i++) {
-				Binding binding = toCopy.getBinding(this.bindingNames[i]);
-				if (binding != null) {
-					this.values[i] = binding.getValue();
-					this.whichBindingsHaveBeenSet[i] = true;
-				}
-			}
-		} else {
-			LinkedHashSet<String> newNames = new LinkedHashSet<>();
-			newNames.addAll(names);
-			newNames.addAll(toCopyBindingNames);
-			this.bindingNames = newNames.toArray(new String[0]);
-			this.whichBindingsHaveBeenSet = new boolean[this.bindingNames.length];
-			this.values = new Value[this.bindingNames.length];
-			for (int i = 0; i < this.bindingNames.length; i++) {
-				Binding binding = toCopy.getBinding(this.bindingNames[i]);
-				if (binding != null) {
-					this.values[i] = binding.getValue();
-					this.whichBindingsHaveBeenSet[i] = true;
```

Comment:
My understanding is that the ArrayBindingSet only supports binding names that are predefined. So if we create an ArrayBindingSet from another BindingSet we can safely ignore all other bindings.